### PR TITLE
fix(VAutocomplete/VCombobox/VSelect): return-object comparison

### LINKED ISSUE #16046: [Bug Report][3.0.0] V-Autocomplete pre selected option not working when using return-object prop
### Environment
**Vuetify Version:** 3.0.0
**Vue Version:** 3.2.41
**Browsers:** Firefox 106.0
**OS:** Ubuntu undefined

### Steps to reproduce
Click v-autocomplete field and see there is no option pre selected while vmodel isnt empty.
Remove return object prop and see it working

### Expected Behavior
Make pre selected option work properly when using filled vmodel and return object prop

### Actual Behavior
Pre selected option doest not work when using filled vmodel and return object prop

### Reproduction Link
[https://play.vuetifyjs.com/#...](https://play.vuetifyjs.com/#eNqNkctuwjAQRX9l5A2tBOHRrmiohNS/wCxMMqiuHNuyJ2wQ/96xQ9JEohKziDJ37jxycriKvffFpUWxFWWsgvYEEan1n9LqxrtAcIWAZ7jBObgGZmydSSstQOVsJKgVKdgly8shqdf0APDB+fUWpCBNBtdSzP/0TdIvyrRJT/ItFx92bv7pZL3vPL5+pIO6cyIarAjrr2fOenY4gLTlsqPDXDghbLxRhJwBlJeF8glYmsRJo7S9Z52gWnKV4w4kHHQOtroazU6K8dnd8j62mndFtiTQ09Ikqm/t47ga+C8Gu3CnH548Lhh1yjv3MbpK80fU0EYM09lp6yJDYmem9qCeYd3rd2Z9NK0hzR88cCiXUw49LpYHXtmTSZbLEWFxm4u3YlO8r0V6WRUrcfwFOI3RcA==)

### Other comments
Thanks Andrew for your help on discord. He said himself its a bug

### PR DESCRIPTION
**Pull request by vexleet**
In the selections computed property the values were checked by using strict equality (triple =), because of that it couldnt find the correct items and we didnt have preselected options. I added deepEqual from @utils in order to properly check for equality between the items.

fix #16046

## How Has This Been Tested?

Created a unit test and also tested it by using the provided code from the issue

## Markup:

<details>

```vue
// Paste your FULL Playground.vue here
```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [ ] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

**Review by johnleider — CHANGES_REQUESTED** [review]
Looks good. This should be added to select and combobox as well.

**Comment by vexleet** [comment]
> Looks good. This should be added to select and combobox as well.

Added the comparison for `Combobox` and for `Select`, currently writing a bit more tests both with objects and strings. I will push the changes tomorrow.

File: packages/vuetify/src/components/VSelect/__tests__/VSelect.spec.cy.tsx [hunk 1]
```diff
@@ -1,6 +1,8 @@
+/// <reference types="..\..\..\..\types\cypress" />
```
Comment (johnleider) [inline packages/vuetify/src/components/VSelect/__tests__/VSelect.spec.cy.tsx:1]:
```suggestion
```

**Comment by vexleet** [comment]
@johnleider can we do these changes using Github commit suggestion? I cant do it locally because when `lint` is run I get errors and can't commit my changes. These lines were added when I ran `lint:fix`

**Comment by KaelWD** [comment]
You can use `--no-verify` to push with lint errors

File: packages/vuetify/src/components/VSelect/VSelect.tsx [hunk 2]
```diff
@@ -91,6 +91,10 @@ export const VSelect = genericComponent<new <
   name: 'VSelect',
 
   props: {
+    valueComparator: {
+      type: Function as PropType<typeof deepEqual>,
+      default: deepEqual,
+    },
```
Comment (KaelWD) [inline packages/vuetify/src/components/VSelect/VSelect.tsx:97]: This can go in makeSelectProps can't it?

Comment (vexleet) [reply packages/vuetify/src/components/VSelect/VSelect.tsx]: Yes! Good point, I added it.

**Pull request merged**